org.eclipse.scout.rt.platform.test/src/test/java/org/eclipse/scout/rt/platform/util/concurrent/AdjustableSemaphoreTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2010-2023 BSI Business Systems Integration AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ */
+package org.eclipse.scout.rt.platform.util.concurrent;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class AdjustableSemaphoreTest {
+
+  @Test
+  public void testCreateInstance_invalidInputs() {
+    assertThrows(IllegalArgumentException.class, () -> new AdjustableSemaphore(0));
+    assertThrows(IllegalArgumentException.class, () -> new AdjustableSemaphore(-1));
+    assertThrows(IllegalArgumentException.class, () -> new AdjustableSemaphore(-10));
+
+    assertThrows(IllegalArgumentException.class, () -> new AdjustableSemaphore(0, true));
+    assertThrows(IllegalArgumentException.class, () -> new AdjustableSemaphore(-1, true));
+    assertThrows(IllegalArgumentException.class, () -> new AdjustableSemaphore(-10, true));
+  }
+
+  @Test
+  public void testGetMaximumPermits() {
+    assertEquals(1, new AdjustableSemaphore(1).getMaximumPermits());
+    assertEquals(10, new AdjustableSemaphore(10).getMaximumPermits());
+    assertEquals(100, new AdjustableSemaphore(100).getMaximumPermits());
+  }
+
+  @Test
+  public void testSetMaximumPermits_invalidInputs() {
+    final AdjustableSemaphore adjustableSemaphore = new AdjustableSemaphore(5);
+
+    assertThrows(IllegalArgumentException.class, () -> adjustableSemaphore.setMaximumPermits(0));
+    assertThrows(IllegalArgumentException.class, () -> adjustableSemaphore.setMaximumPermits(-1));
+    assertThrows(IllegalArgumentException.class, () -> adjustableSemaphore.setMaximumPermits(-10));
+  }
+
+  @Test
+  public void testSetMaximumPermits() {
```

Comment:
Could you also add some test cases that change the maximumNumber of permits while other parties have already acquired some or all of them before. Something like:

```
final AdjustableSemaphore adjustableSemaphore = new AdjustableSemaphore(5);
assertEquals(5, adjustableSemaphore.getMaximumPermits());
assertTrue(adjustableSemaphore.tryAcquire());
assertTrue(adjustableSemaphore.tryAcquire());
assertTrue(adjustableSemaphore.tryAcquire());
// now 3 of 5 permits are in use

adjustableSemaphore.setMaximumPermits(3);
assertEquals(3, adjustableSemaphore.getMaximumPermits());
assertFalse(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release();
assertTrue(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release(2);
// now 1 of 3 permits are in use

adjustableSemaphore.setMaximumPermits(2);
assertTrue(adjustableSemaphore.tryAcquire());
assertFalse(adjustableSemaphore.tryAcquire());
// now 2 of 2 permits are in use



// maybe another test case
final AdjustableSemaphore adjustableSemaphore = new AdjustableSemaphore(5);
assertTrue(adjustableSemaphore.tryAcquire(5));

adjustableSemaphore.setMaximumPermits(2);
// now 5 of 2 permits are in use
assertFalse(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release();
// 4 of 2 permits are in use
assertFalse(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release(2);
// 2 of 2 permits are in use
assertFalse(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release();
// 1 of 2 permits are in use
assertTrue(adjustableSemaphore.tryAcquire());
```

Reply:
Good idea. Added two additional test cases for increasing/decreasing the permits.